cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -276,3 +381,94 @@ def _from_json_dict_(cls, metadata, **kwargs):
 
     def _value_equality_values_(self):
         return self._metadata
+
+
+def _set_gate_in_gate_spec(
+    gate_spec: v2.device_pb2.GateSpecification, gate_family: cirq.GateFamily
+) -> None:
+    if gate_family == cirq.GateFamily(ops.SYC):
+        gate_spec.syc.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.SQRT_ISWAP):
+        gate_spec.sqrt_iswap.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.SQRT_ISWAP_INV):
+        gate_spec.sqrt_iswap_inv.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.CZ):
+        gate_spec.cz.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.PhasedXZGate):
+        gate_spec.phased_xz.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.ZPowGate, tags_to_ignore=[ops.PhysicalZTag()]):
+        gate_spec.virtual_zpow.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.ZPowGate, tags_to_accept=[ops.PhysicalZTag()]):
+        gate_spec.physical_zpow.SetInParent()
+    elif gate_family == cirq.GateFamily(experimental_ops.CouplerPulse):
+        gate_spec.coupler_pulse.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.MeasurementGate):
+        gate_spec.meas.SetInParent()
+    elif gate_family == cirq.GateFamily(cirq.WaitGate):
+        gate_spec.wait.SetInParent()
+    else:
+        raise ValueError(f'Unrecognized gate {gate_family}.')
+
+
+def to_proto(
```

Comment:
I think this should have a more descriptive name if it is to be a free function.  Otherwise, we couldn't import it into the global namespace, for instance, since there are many objects that could be protos.  Maybe `create_device_specification_proto`